test/llvm-intrinsics/lifetime.ll

```diff
@@ -6,14 +6,45 @@
 ; RUN: llvm-spirv -r %t.spv -o %t.spv.bc
 ; RUN: llvm-dis < %t.spv.bc | FileCheck %s --check-prefix=CHECK-LLVM
 
-; CHECK-SPIRV: 3 LifetimeStart [[tmp:[0-9]+]] 0
-; CHECK-SPIRV: 3 LifetimeStop [[tmp]] 0
+; CHECK-SPIRV-DAG: Name [[#SimpleF:]] "lifetime_simple"
+; CHECK-SPIRV-DAG: Name [[#SizedF:]] "lifetime_sized"
+; CHECK-SPIRV-DAG: Name [[#GenericF:]] "lifetime_generic"
 
+; CHECK-SPIRV: Function [[#]] [[#SimpleF:]]
+; CHECK-SPIRV: LifetimeStart [[#Tmp:]] 0
+; CHECK-SPIRV: LifetimeStop [[#Tmp]] 0
+
+; CHECK-SPIRV: Function [[#]] [[#SizedF:]]
+; CHECK-SPIRV: LifetimeStart [[#Tmp:]] 1
+; CHECK-SPIRV: Bitcast [[#]] [[#Cast:]]
+; CHECK-SPIRV: LifetimeStop [[#Cast]] 1
+
+; CHECK-SPIRV: Function [[#]] [[#GenericF:]]
+; CHECK-SPIRV: Variable [[#]] [[#Var:]] 7
+; CHECK-SPIRV: PtrCastToGeneric [[#]] [[#Cast1:]] [[#Var]]
```

Comment:
potentially extra analysis might be done removing useless (in this case) casts, do we want it? probably such analysis wouldn't be generic